bridgerelayer/chain/substrate/chain.go

```diff
@@ -0,0 +1,102 @@
+package substrate
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/sirupsen/logrus"
+	"github.com/snowfork/polkadot-ethereum/bridgerelayer/chain"
+	"github.com/snowfork/polkadot-ethereum/bridgerelayer/crypto/sr25519"
+	"github.com/spf13/viper"
+)
+
+// Chain ...
```

Comment:
Makes sense to add proper method descriptions in this PR

Reply:
Well actually, what I've been working towards is actually not having comments like that. In the latest upload I've removed some redundant filler comments.

I'm now using [Revive](https://github.com/mgechev/revive) for linting. This a configurable linter that allows us to disable the annoying "Exported thing must have a comment or be unexported". I don't particularly like that rule as I don't believe it leads to better or more useful comments. Our packages are all being used internally and are not being published, so names are not really being exported in the full sense of the word.

Plus we're going to rewrite parts of the relay substantially, and having to add meaningless comments for exports will just add more work.

Reply:
That's reasonable. In this context, I agree that the comments are somewhat meaningless. Will it raise any concerns during the client's code review? Can probably get around that if it comes up by explaining the thought process here.